src/cloudant/index.py

```diff
@@ -262,3 +262,91 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
+    """
+    Encapsulates a SearchIndex as a dictionary based object, exposing the
+    search index function and analyzer as attributes.
+    """
+    def __init__(
+            self,
+            ddoc,
+            search_index_name=None,
+            search_index=None
+    ):
+        super(SearchIndex, self).__init__()
+        self.design_doc = ddoc
+        if search_index_name is not None:
+            self._search_index_name = search_index_name
+        if search_index is not None:
+            self['index'] = codify(search_index)
+        self['analyzer'] = codify('standard')
+
+    @property
+    def analyzer(self):
+        """
+        Provides an search index analyzer property accessor and setter.
+
+        For example:
+
+        .. code-block:: python
+
+            # Set the SearchIndex index property
+            search.analyzer = 'standard'
+
+            print search.analyzer
+
+        :param str index_func: Javascript search index function.
+
+        :returns: Codified search index function
+        """
+        return self.get('analyzer')
+
+    @analyzer.setter
+    def analyzer(self, analyzer=None):
+        """
+        Provides a search index analyzer property setter.
+        Default settings is a 'standard' analyzer.
+        """
+        if isinstance(analyzer, STRTYPE) and analyzer in SEARCH_ANALYZERS:
+            self['analyzer'] = codify(analyzer)
+        else:
+            raise CloudantArgumentError('The analyzer property must be a '
+                                        'valid string and supported by '
+                                        'Cloudant search')
+
+    @property
+    def index(self):
+        """
+        Provides an search index property accessor and setter.
+
+        For example:
+
+        .. code-block:: python
+
+            # Set the SearchIndex index property
+            search.index = 'function (doc) {\n  index(\"default\", doc._id);\n}'
```

Comment:
Removed print in code block: https://github.com/cloudant/python-cloudant/pull/169/files#diff-eeec48d80ce666f16517361d09cf63f1R329